### PR TITLE
Ft add email verification and password reset

### PR DESCRIPTION
**Pull request by MutegekiHenry**
PR Checklist

## Description

- It adds three endpoints to perform user account verification and password reset via email. 

Pull branch and test the functionality in the following endpoints
- /users/reset_password/{token}
- /users/verify/{token}
- /users/verify

Add the Email credentials to your .env file 

APP_MAIL_USERNAME=epilepsy.smartapp@gmail.com
APP_MAIL_PASSWORD=

Install this requirement in your environment: Flask-Mail==0.9.1

### CODE REVIEW

**Comment by MutegekiHenry** [comment]
Password on slack

File: app/routes/__init__.py
Comment (Schade9) [inline app/routes/__init__.py:1]: Some imports are missing in this file. `EmailVerificationRequest`.
`ResetPassword` should change to `ResetPasswordView`

File: app/controllers/__init__.py
Comment (Schade9) [inline app/controllers/__init__.py:1]: Change `ResetPassword` to `ResetPasswordView`
Add `UserEmailVerificationView` and `EmailVerificationRequest` imports

File: app/controllers/user.py
Comment (Schade9) [inline app/controllers/user.py:1]: Was VERIFICATION_SALT and MAIL_DEFAULT_SENDER added to the `config.py` file?
We have an error for verification salt when registering a user. line 57.

**Review by Schade9 — CHANGES_REQUESTED** [review]
I have noticed some changes that have to be made.

**Review by Schade9 — APPROVED** [review]

**Pull request merged**